src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -795,33 +791,23 @@ public SkyValue compute(SkyKey key, Environment env) throws ConfiguredTargetFunc
       throw e;
     }
 
-    Map<Label, ConfigMatchingProvider> configConditions = new LinkedHashMap<>();
+    Map<Label, ConfiguredTargetAndData> asConfiguredTargets = new LinkedHashMap<>();
```

Comment:
Ideally this entire method, or at least the part from here on, could be moved into ConfigConditions.

Reply:
I support this idea, and tried to make headway with it. But it's too complicated for this PR, in my view. The problem is that the method has various dependencies on ConfiguredTargetFunction that aren't trivial to extract out. Trivially moving it would result in ConfiguredTargetFunction and ConfigConditions calling back and forth to each other.

I do support keeping ConfiguredTargetFunction (which is a beast) simple in any ways we can. So over a longer term I'd love to make progress on things like this.